app/src/main/java/com/arctouch/io/outdoorsychallenge/features/searchrv/adapters/SearchRvVehicleViewModel.kt

```diff
@@ -1,10 +1,20 @@
 package com.arctouch.io.outdoorsychallenge.features.searchrv.adapters
 
 import androidx.lifecycle.ViewModel
+import com.arctouch.io.outdoorsychallenge.R
 import com.arctouch.io.outdoorsychallenge.domain.model.Vehicle
 
 class SearchRvVehicleViewModel(vehicle: Vehicle) : ViewModel() {
 
+    val id = vehicle.id
     val name = vehicle.name
     val imageUrl = vehicle.imageUrl
+    val favoriteResource = if (vehicle.isFavorite)
+        R.drawable.icon_star_selected
+    else
+        R.drawable.icon_star_unselected
```

Comment:
A viewModel is a peace of logic that doesn't know anything about android details, so you shouldn't add the resources here, you can do it by binding, checking on the xml side if isFavorite is true or false and adding to it the correct icon resource